integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/status/DefaultConversationService.java

```diff
@@ -49,23 +50,15 @@
     private final IpMailSender ipMailSender;
     private final Clock clock;
     private final StatusQueue statusQueue;
-    private final ObjectProvider<MeldingsformidlerClient> meldingsformidlerClientObjectProvider;
-    @Getter(lazy = true) private final Set<ServiceIdentifier> pollables = createrPollables();
+    private final ObjectProvider<StatusStrategy> statusStrategyProvider;
+    @Getter(lazy = true) private final Map<ServiceIdentifier, StatusStrategy> statusStrategyMap = createStatusStrategyMap();
 
     private static final Set<ReceiptStatus> COMPLETABLES = Sets.newHashSet(LEST, FEIL, LEVETID_UTLOPT, INNKOMMENDE_LEVERT);
 
-    private Set<ServiceIdentifier> createrPollables() {
-        Set<ServiceIdentifier> serviceIdentifiers = new HashSet<>();
-        serviceIdentifiers.add(DPV);
-        serviceIdentifiers.add(DPF);
-
-
-        if (meldingsformidlerClientObjectProvider.stream()
-                .anyMatch(MeldingsformidlerClient::skalPolleMeldingStatus)) {
-            serviceIdentifiers.add(DPI);
-        }
-
-        return Collections.unmodifiableSet(serviceIdentifiers);
+    private Map<ServiceIdentifier, StatusStrategy> createStatusStrategyMap() {
+        return statusStrategyProvider.stream().collect(
+                Collectors.toMap(StatusStrategy::getServiceIdentifier, Function.identity())
```

Comment:
Blir dette eit map som tåler samtidige oppslag?

Reply:
Usikker - Jeg kan ordne så den bruker ConcurrentHashMap....men den er jo bare read-only, så skulle tro det går greit uansett`?

Reply:
Eg må alltid dobbeltsjekke, kan vi legge inn concurrenthashmap for sikkerhets skuld?